packages/text/src/Text.ts

```diff
@@ -503,8 +510,13 @@ export class Text extends Sprite
                     {
                         stop = currentIteration / totalIterations;
                     }
-                    gradient.addColorStop(stop, fill[j]);
+
+                    // Prevent color stop generation going backwards from floating point imprecision
+                    const clampedStop = Math.max(lastIterationStop, stop);
```

Comment:
do we also want to make sure that we don't have a point that's greater than 1? Or do you feel that's not possible to achieve even with the rounding issues?

Reply:
Trying to add a stop >1 does throw a DOMException from addColorStop apparently, so might be a good idea for safety's sake.